source/draw/gpu/opengl/AndroidContext.ooc

```diff
@@ -146,10 +148,23 @@ AndroidContext: class extends OpenGLContext {
 		map add("startY", startY)
 		DrawState new(target) setMap(map) draw()
 	}
-	preallocate: override func (resolution: IntVector2D) { GraphicBufferYuv420Semiplanar free~all() }
+	preallocate: override func (resolution: IntVector2D) { this _eglBin clear() }
 	preregister: override func (image: Image) {
 		if (image instanceOf(GraphicBufferYuv420Semiplanar))
-			(image as GraphicBufferYuv420Semiplanar) toRgba(this) referenceCount decrease()
+			this createEGLRgba(image as GraphicBufferYuv420Semiplanar) referenceCount decrease()
 	}
+	createEGLRgba: func (source: GraphicBufferYuv420Semiplanar) -> EGLRgba {
+		result := this _eglBin search(|image| source buffer _handle == image buffer _handle)
+		if (result == null) {
+			padding := source uvOffset - source stride * source size y
+			extraRows := padding align(source stride) / source stride
+			height := source size y + source size y / 2 + extraRows
+			width := source stride / 4
+			rgbaBuffer := source buffer shallowCopy(IntVector2D new(width, height), width, GraphicBufferFormat Rgba8888, GraphicBufferUsage Texture | GraphicBufferUsage RenderTarget)
+			result = EGLRgba new(rgbaBuffer, this)
+		}
```

Comment:
Previously we had to make sure the reference counter was always >= 1 so that the `EGLRgba` was never freed. Now we always want it to be freed since we moved the recycling to the `free` function in `EGLRgba`.